File: Arrays & Strings/002 Strings/Valid Anagram/Solution.java
import java.util.Arrays;

class Solution {
    public boolean isAnagram(String s, String t) {
        int len=s.length();
        if(len!=t.length())
            return false;
        
        int count[] = new int[26];
        
        //Add count to array
        for(int i=0;i<len;i++)
            count[s.charAt(i)-'a']++;
        
        //Subtract count from array
        for(int i=0;i<len;i++)
            count[t.charAt(i)-'a']--;
        
        //Check if addition=subtraction
        for(int i:count)
            if(i!=0)
                return false;
        
        return true;
    }
}

File: Arrays & Strings/002 Strings/Valid Anagram/Solution.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        val len=s.length
        if(len!=t.length)
            return false
        
        val count= IntArray(26)
        
        //Add count to array
        for(c in s)
            count[c-'a']++
        
        //Subtract count from array
        for(c in t)
            count[c-'a']--
        
        //Check if addition=subtraction
        for(i in count)
            if(i!=0)
                return false
        
        return true
    }
}
